### DISCUSSION TITLE
[Miscellaneous] NO public API for "Get a list of 'ERC1155 - Token Transfer Events' by Address."

### DISCUSSION THREAD

**Discussion by opfocus** [ethereum-optimism/developers#48] [Miscellaneous] [answered]
### Issue Description

It seems that the op-mainnet etherscan does not have a public API for "Get a list of 'ERC1155 - Token Transfer Events' by Address."
[op-mainnet API](https://docs.optimism.etherscan.io/api-endpoints/accounts)
[ethereum API](https://docs.etherscan.io/api-endpoints/accounts)

**Comment by sbvegan** [comment]
Hey @opfocus, I'll ask the etherscan team and see if they support it. I'll let you know what I find out.

**Reply by sbvegan** [reply, marked as answer]
Wow they work fast! They just implemented the endpoint. They said it might be a little less optimized than the other token endpoints, but they said they could roll out some optimizations in the coming days

http://api-optimistic.etherscan.io/api?module=account&action=token1155tx&address=0x57317a302f8874527c5d4781993ea56814315b1c&sort=asc&apikey=YourApiKeyToken
